apis/r/src/rutilities.cpp

```diff
@@ -544,3 +544,50 @@ const char *_tiledb_layout_to_string(tiledb_layout_t layout) {
         Rcpp::stop("unknown tiledb_layout_t (%d)", layout);
     }
 }
+
+// internal helper function for
+// `_get_filter_options()`
+// <Numeric> should be `int32_t` or `double`
+template <typename Numeric>
+Numeric _get_filter_option(Rcpp::XPtr<tiledb::Filter> filter, tiledb_filter_option_t option) {
+    try {
+        switch (option) {
+        case TILEDB_BIT_WIDTH_MAX_WINDOW:
+        case TILEDB_POSITIVE_DELTA_MAX_WINDOW:
+            return static_cast<Numeric>(filter->get_option<uint32_t>(option));
+        case TILEDB_SCALE_FLOAT_BYTEWIDTH:
+            return static_cast<Numeric>(filter->get_option<uint64_t>(option));
+        default:
+            return filter->get_option<Numeric>(option);
+        }
+    } catch (const tiledb::TileDBError& e) {
+        const std::string msg = e.what();
+        if (msg.find("unknown option") != std::string::npos) {
+            if (std::is_same<Numeric, double>::value) {
+                return R_NaReal;
+            }
+            return R_NaInt;
+        }
+        if (msg.find("Filter does not support options") != std::string::npos) {
+            if (std::is_same<Numeric, double>::value) {
+                return R_NaReal;
+            }
+            return R_NaInt;
+        }
+        throw(e);
+    }
+}
```

Comment:
@mojaveazure and I have already discussed this, but I strongly believe try/catch should be avoided when possible except when re-throwing. I think we should push on the C++ API to add the functions we need to do this in a safer way.

Reply:
@jp-dark I completely agree with you regarding behavior of an ideal API.

We need to call the core API we have, working around its shortcomings, request an improved API in some upcoming core release, and do the best we can (which is this) until then.

Reply:
@jp-dark @johnkerl I'm going to land this PR now as:
- core doesn't have a safer API to use
- this isn't going directly to `main` but into a separate branch, so it should be easy to replace once a safer API exists

I've filed [SC-65200](https://app.shortcut.com/tiledb-inc/story/65200) to track adopting a safer API once it exists